api/v1/candidates.coffee

```diff
@@ -1 +1,47 @@
+ObjectID = require('mongodb').ObjectID
+
 module.exports = (api, db) ->
+  api = api || {}
+  api.v1 = api.v1 || {}
+
+  api.v1.getCandidate = (req, res) ->
+    db.collection("candidates").findOne( {_id: new ObjectID(req.params.id)}, (err, results) ->
+      if err
+        res.jsonp
+          error: err
+        , 500
+
+        return
+
+      res.jsonp(results)
+    )
+
+  api.v1.getCandidates = (req, res) ->
+    api.v1.search(req, res)
+
+  api.v1.search = (req, res) ->
+    findCriteria = req.query || {}
+    if findCriteria.nameId
+      findCriteria.nameId = parseInt(findCriteria.nameId) || findCriteria.nameId
+    db.collection("candidates").find(findCriteria).toArray (err, results) ->
```

Comment:
Yeah, so this is according to the spec. What we should do is take in a `firstName` and a `lastName` parameter. If the parameter is defined we do a search based on it. If not, it's omitted. So, to do this, we'll want to build the MongoDB query incrementally, like so:

```javascript
queryParts = {}

for term in ['firstName', 'lastName']
  if findCriteria[term]?
    queryParts[term] = findCriteria[term]

db.collection("candidates").find({'$and': queryParts})...
```